### ISSUE TITLE
Adding support for cookie-session broke express-session save() logic

### ISSUE THREAD

**Issue by KidkArolis** [simov/grant#146]
Described in context here: https://github.com/simov/grant/commit/d39902b43520d5f67db1f74b451b8b897c92e7b5#r36073853

Haven't come up with a possible fix yet

**Comment by KidkArolis** [comment]
As a workaround, I'm currently doing this in my app:

```js
  function expressSessionWithFixedSave(req, res, next) {
    expressSession(req, res, function onNext(err) {
      if (err) return next(err)
      const _save = req.session.save
      req.session.save = function(cb) {
        _save.apply(req.session, [cb])
      }
      next()
    })
  }
```

But obviously, would be great to fix in `grant`.

**Comment by simov** [comment]
I think this should do it:

```js
var redirect = (req, res, url) =>
  typeof req.session.save === 'function' &&
    Object.getPrototypeOf(req.session).save.length
    ? req.session.save(() => res.redirect(url))
    : res.redirect(url)
```

It checks the prototype for the presence of a callback instead of the instance.

Can you check that out on your end? It seems to be working with my local example. Otherwise I'm going to push it to the next release.

**Comment by simov** [comment]
The fix was published in v4.6.6, let me know if it works for you.
